x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTimeToCharProcessorTests.java

```diff
@@ -0,0 +1,322 @@
+            "GMT", "UTC", "CET",
+            // offsets
+            "+11:00", "+04:30", "+01:00", "+00:00", "-00:00", "-01:15", "-02:00", "-11:00");
+
+        private final int id;
+        private final BigDecimal secondsAndFractionsSinceEpoch;
+        private final String formatString;
+        private final String zoneId;
+        
+        TestCase(BigDecimal secondsAndFractionsSinceEpoch, String formatString) {
+            this.id = COUNTER.incrementAndGet();
+            this.secondsAndFractionsSinceEpoch = secondsAndFractionsSinceEpoch;
+            this.formatString = formatString;
+            this.zoneId = TestGenerator.randomFromCollection(POSTGRES_TEST_ZONE_LIST);
+        }
+
+    }
+
+    /**
+     * Generates an sql file that can be used to generate the test dataset for this unit test (to be used with the <code>psql</code> cli).
+     *
+     * Run the @{link {@link TestGenerator#main(String[])}} class and execute the following command to generate the output dataset with:
+     *
+     * <p>
+     *  <code>
+     *      # easy way to spin up the latest Postgres
+     *      docker run --rm --name postgres-latest -e POSTGRES_PASSWORD=mysecretpassword -p 5432:5432 -d postgres:latest
+     *
+     *      # generate the csv for the unit test
+     *      PGPASSWORD="mysecretpassword" psql --quiet -h localhost -p 5432 -U postgres -f /tmp/postgresql-tochar-test.sql \
+     *          &gt; /path/to/tochar.csv
+     *  </code>
+     * </p>
+     */
+    private static class TestGenerator {
+        
+        private static final long SECONDS_IN_YEAR = 365 * 24 * 60 * 60L;
+        private static final String DELIMITER = "|";
+        private static final String PATTERN_DELIMITER = " @ ";
+        // these patterns are hard to sync up between PostgreSQL and Elasticsearch, so we just warn, but actually
+        // accept the output of Elasticsearch as is
+        private static final Set<String> NOT_FULLY_MATCHABLE_PATTERNS = Set.of("TZ", "tz");
+        private static final Set<String> UNSUPPORTED_PATTERN_MODIFIERS = Set.of("FX", "TM", "SP");
+        private static final String[] PATTERNS = new String[] {
+            "HH", "HH12", "HH24", "MI", "SS", "MS", "US", "FF1", "FF2", "FF3", "FF4", "FF5", "FF6", "SSSS", "SSSSS", "AM", "am", "PM",
+            "pm", "A.M.", "a.m.", "P.M.", "p.m.", "Y,YYY", "YYYY", "YYY", "YY", "Y", "IYYY", "IYY", "IY", "I", "BC", "bc", "AD", "ad",
+            "B.C.", "b.c.", "A.D.", "a.d.", "MONTH", "Month", "month", "MON", "Mon", "mon", "MM", "DAY", "Day", "day", "DY", "Dy",
+            "dy", "DDD", "IDDD", "DD", "D", "ID", "W", "WW", "IW", "CC", "J", "Q", "RM", "rm", "TZ", "tz", "TZH", "TZM", "OF"
+        };
+        private static final List<String> FILL_MODIFIERS = List.of("FM", "fm", "");
+        private static final List<String> ORDINAL_SUFFIX_MODIFIERS = List.of("TH", "th", "");
+
+        @SuppressForbidden(reason = "It is ok to use ThreadLocalRandom outside of an actual test")
```

Comment:
Can you expand?

Reply:
Instead of calling the test suite Random externally, which is unsupported since there are no guarantees about its initialization, simply the `ESTestCase`.